bool softMode
uint8 forceLevel
bool gripInside
---
bool forceModeChanged
bool gripDirectionChanged